src/api/db/models/Image.js

```diff
@@ -316,6 +322,82 @@ export class ImageModel {
     }
   }
 
+  static async deleteComment(input, context) {
+    try {
+      const image = await ImageModel.queryById(input.imageId, context);
+
+      const comment = (image.comments || []).filter((c) => { return c._id === input.id; })[0];
+      if (!comment) throw new Error('Comment not found on image');
+
+      if (comment.author !== context.user['cognito:username'] && !context.user['is_superuser']) {
+        throw new Error('Can only edit your own comments');
+      }
+
+      image.comments = image.comments.filter((c) => { return c._id !== input.id; });
+
+      await image.save();
+
+      return { message: 'Images Deleted' };
+    } catch (err) {
+      // if error is uncontrolled, throw new ApolloError
+      if (err instanceof ApolloError) throw err;
+      throw new ApolloError(err);
+    }
+  }
+
+  static async updateComment(input, context) {
+    try {
+      const image = await ImageModel.queryById(input.imageId, context);
+
+      const comment = (image.comments || []).filter((c) => { return c._id === input.id; })[0];
+      if (!comment) throw new Error('Comment not found on image');
+
+      if (comment.author !== context.user['cognito:username'] && !context.user['is_superuser']) {
+        throw new Error('Can only edit your own comments');
+      }
+
+      comment.comment = input.comment;
+
+      await image.save();
+
+      return comment;
+    } catch (err) {
+      // if error is uncontrolled, throw new ApolloError
+      if (err instanceof ApolloError) throw err;
+      throw new ApolloError(err);
+    }
+  }
+
+  static async createComment(input, context) {
+    const operation = async (input) => {
+      return await retry(async (bail, attempt) => {
+        if (attempt > 1) console.log(`Retrying createComment operation! Try #: ${attempt}`);
+
+        // find images, add comment, and bulk write
+        return await Image.bulkWrite([{
```

Comment:
I don't have a strong opinion on this, but I don't think we need to necessarily use `bulkWrite()` here. The reason why I refactored the `updateObjects()` and `upateLabels()` resolvers to use `bulkWrite()` was to support the ability for the frontend to make calls to the API to perform multpile CRUD operations on `image.objects` at once (see PR #127), which I don't think we need to support for image comments.

So this obviously works, but I think we could also follow the `ImageModel.queryById`, make the update, then `image.save()` pattern if there was any advantage to that.